src/LanguageServer/Impl/Sources/DefinitionSource.cs

```diff
@@ -98,26 +112,41 @@ internal sealed class DefinitionSource {
                         if (member is IPythonInstance v) {
                             return new Reference { range = v.Location.Span, uri = v.Location.DocumentUri };
                         }
-                        return await FromMemberAsync(member, null, eval, cancellationToken);
+                        return await FromMemberAsync(member, null, statement, analysis, cancellationToken);
                     }
             }
 
-            if (node != null && CanNavigateToModule(module) && module is IDocument doc) {
+            module = ToRealModule(module, analysis);
+            if (node != null && CanNavigateToModule(module, analysis) && module is IDocument doc) {
                 return new Reference {
-                    range = node.GetSpan(doc.GetAnyAst()), uri = doc.Uri
+                    range = location?.Span ?? node.GetSpan(doc.GetAnyAst()), uri = doc.Uri
                 };
             }
 
             return null;
         }
 
-        private static bool CanNavigateToModule(IPythonModule m) {
+        private static bool CanNavigateToModule(IPythonModule m, IDocumentAnalysis analysis) {
+            if (m.Uri != null) {
+                var fs = analysis.ExpressionEvaluator.Services.GetService<IFileSystem>();
+                if (!fs.FileExists(m.Uri.AbsolutePath)) {
```

Comment:
Do we still need `UriExtensions.ToAbsolutePath`?

Reply:
I probably should be using `ToAbsolutePath`